v2/src/log_context_property_basic_types.c

```diff
@@ -13,149 +14,209 @@
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_042: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int16_t).to_string shall call snprintf with buffer, buffer_length and format string PRId16 and pass in the values list the int16_t value pointed to be property_value. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_043: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int16_t).to_string shall succeed and return the result of snprintf. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_044: [ If any error is encountered, LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int16_t).to_string shall fail and return a negative value. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_052: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint16_t).to_string shall call snprintf with buffer, buffer_length and format string PRId16 and pass in the values list the uint16_t value pointed to be property_value. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_053: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint16_t).to_string shall succeed and return the result of snprintf. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_054: [ If any error is encountered, LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint16_t).to_string shall fail and return a negative value. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_062: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int8_t).to_string shall call snprintf with buffer, buffer_length and format string PRId8 and pass in the values list the int8_t value pointed to be property_value. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_063: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int8_t).to_string shall succeed and return the result of snprintf. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_064: [ If any error is encountered, LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int8_t).to_string shall fail and return a negative value. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_072: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint8_t).to_string shall call snprintf with buffer, buffer_length and format string PRId8 and pass in the values list the uint8_t value pointed to be property_value. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_073: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint8_t).to_string shall succeed and return the result of snprintf. ]*/ \
+            /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_074: [ If any error is encountered, LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint8_t).to_string shall fail and return a negative value. ]*/ \
+            result = snprintf(buffer, buffer_length, "%" print_format_string "", *(type_name*)property_value); \
         } \
         return result; \
     } \
 
-#define DEFINE_BASIC_TYPE_COPY(type_name) \
-    static int MU_C2(type_name,_log_context_property_copy)(void* dst_value, void* src_value) \
+#define DEFINE_BASIC_TYPE_COPY(type_name, print_format_string) \
```

Comment:
>print_format_string [](http://example.com/codeflow?start=42&length=19)

why is `print_format_string` required for `TYPE_COPY`? It doesn't look like it is used...

Reply:
Not needed, removed.